packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -0,0 +1,241 @@
+
+#### Alternative 1: Use `table-column-text`
+
+With the changes proposed in [HLD for programmatically sorting columns](https://github.com/ni/nimble/pull/1049) to allow a column to be sorted by a different data field than the one being used for display, many of the above use cases could be met with minor changes to the existing text column. Clients would write custom logic to populate their data with a new string field that contains formatted values. Then they would configure the table to display that string field while sorting by the original numeric field.
+
+```html
+<nimble-table>
+    <nimble-table-column-text
+        operational-data-field-name="progress"
+        field-name="formattedProgress"
+    >
+        Progress
+    <nimble-table-column-text>`
+</nimble-table>
+```
+
+```ts
+const originalData = [{ progress: 0.1 }, { progress: 0.2 }];
+const tableData = originalData.map(x => {
+    progress: x.progress;
+    formattedProgress: x ? `${100 * x.progress}%` : undefined;
+});
+table.data = tableData;
+```
+
+**Pros:**
+
+-   formatted data is specified up front, guaranteeing fast scroll performance
+-   powerful; clients can format data however they want, including via browser APIs which are i18n-friendly or via server-side logic
+
+**Cons:**
+
+-   Increased memory usage and data update time from clients pre-populating data with field for each formatted column
+-   Added complexity of writing procedural code even for simple formatting use cases
+-   Potential cross-app inconsistency if formatting code isn't shared
+
+**Implementation Cost:**
+
+-   Exposing `operational-data-field-name` to be set by client code rather than column definition
+
+#### Alternative 2: Client specifies formatting function
+
+When configuring a column, clients could provide a callback function that converts data of any supported type into a formatted string.
+
+There isn't a good way to set a function as an attribute value on a column, so the function would be specified in JS code. One possible mechanism would be for clients to override an abstract base class and register a new column type:
+
+```html
+<nimble-table>
+    <my-app-progress-column
+        field-name="progress"
+    >
+        Progress
+    <my-app-progress-column>`
+</nimble-table>
+```
+
+```ts
+class MyAppProgressColumn : NimbleFormattedTextColumnBase<number> {
+    public override format(value: number) : string {
+        return `${100 * value}%`;
+    }
+
+    public override shouldUsePlaceholder(value: number | undefined) : boolean {
+        return value === undefined;
+    }
+}
+
+MyAppProgressColumn.registerColumn('my-app-progress-column');
+```
+
+Some of this is prototyped in the [number-column-prototype branch](https://github.com/ni/nimble/compare/main...number-column-prototype?expand=1).
+
+Other variants of this idea include:
+
+1. setting the formatting function as a property on a column element.
```

Comment:
This could be overly burdensome for the Blazor component.

Reply:
Obviously the potential drawback to this option is the performance penalty that would be incurred by each rendered cell to call into .NET, and then back to JS.

Reply:
After discussion, the current recommendation is that Blazor clients would provide this configuration in JS (even though it might be technically possible to achieve in .NET for some variants of this idea). I clarified this in the Cons section.